acceptance/testdata/repos/repo-archive-unarchive.txtar

```diff
@@ -0,0 +1,26 @@
+# Use gh as a credential helper
+exec gh auth setup-git
+
+# Create a repository with a file so it has a default branch
+exec gh repo create $ORG/$SCRIPT_NAME-$RANDOM_STRING --add-readme --private
+
+# Defer repo cleanup
+defer gh repo delete --yes $ORG/$SCRIPT_NAME-$RANDOM_STRING
+
+# Check that the repo exists and isn't archived
+exec gh repo view $ORG/$SCRIPT_NAME-$RANDOM_STRING --json=isArchived --jq='.isArchived'
+stdout false
```

Comment:
Cross referencing a [conversation](https://github.com/cli/cli/pull/9782#discussion_r1805268802) from @andyfeller's PR:

Paraphrasing @williammartin comment:

> Can you talk to me a bit about why you feel you need to check a [precondition] before proceeding? How's this moving the needle on your confidence levels?

I think for me it was about instilling confidence that the thing I'm going to assert against to define success of a command was indeed changed or affected. It was the loop I did myself in the terminal to gain confidence that the commands were working as expected.

There is an additional effect in that it protects against unintended changes to a command's default behavior. In my example, if, for some reason, `gh repo create` started automatically creating `archived` repos, we'd catch it here. Maybe that's out of scope of my test? 🤔

Reply:
Thanks for the explanation.

I think that "hey script reader, up front here's the thing this specific test cares about changing and I'm communicating that to you" is probably the thing that resonates with me most. That's why in the other PR I suggested that if we did have these precondition checks that they are the same assertion differing only in whether they are expecting or refuting state.

I think that if we wanted to validate that repos are created in an unarchived form, then that should be tested against `repo create` directly, so that the confidence in that command is dependent on some other command that may or may not be testing it.

But I want to be clear here, I'm **not** saying that these should be removed. I am saying that I think it's important that there is a clear and shared pattern to all the testscripts that we can say "make it look like that". That can also mean going back to the scripts that are already being merged and making them use this pre and post pattern.

Reply:
> "hey script reader, up front here's the thing this specific test cares about changing and I'm communicating that to you"

This is definitely my primary motivation for doing it. I hear you on the pattern. That's probably a good chat to have synchronously. Maybe we talk about it in retro?